src/reactive-handler.ts

```diff
@@ -132,62 +113,12 @@ export class ReactiveProxyHandler {
         return getPrototypeOf(originalTarget);
     }
     getOwnPropertyDescriptor(shadowTarget: ReactiveMembraneShadowTarget, key: PropertyKey): PropertyDescriptor | undefined {
-        const { originalTarget, membrane } = this;
-        const { valueObserved } = this.membrane;
-
-        // keys looked up via hasOwnProperty need to be reactive
-        valueObserved(originalTarget, key);
-
-        let desc = getOwnPropertyDescriptor(originalTarget, key);
-        if (isUndefined(desc)) {
-            return desc;
-        }
-        const shadowDescriptor = getOwnPropertyDescriptor(shadowTarget, key);
-        if (!isUndefined(shadowDescriptor)) {
-            return shadowDescriptor;
-        }
-        // Note: by accessing the descriptor, the key is marked as observed
-        // but access to the value, setter or getter (if available) cannot observe
-        // mutations, just like regular methods, in which case we just do nothing.
-        desc = wrapDescriptor(membrane, desc, wrapValue);
-        if (!desc.configurable) {
-            // If descriptor from original target is not configurable,
-            // We must copy the wrapped descriptor over to the shadow target.
-            // Otherwise, proxy will throw an invariant error.
-            // This is our last chance to lock the value.
-            // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/handler/getOwnPropertyDescriptor#Invariants
-            ObjectDefineProperty(shadowTarget, key, desc);
-        }
-        return desc;
+        return getOwnPropertyDescriptorMembraneTrap.call(this, shadowTarget, key);
     }
     preventExtensions(shadowTarget: ReactiveMembraneShadowTarget): boolean {
-        const { originalTarget, membrane } = this;
-        lockShadowTarget(membrane, shadowTarget, originalTarget);
-        preventExtensions(originalTarget);
-        return true;
+        return preventExtensionsMembraneTrap.call(this, shadowTarget);
```

Comment:
abstracted

Reply:
well, we are not doing much optimizations around this... copying again into the shadow target should not be a problem, it is just overhead, but the reality is that must of these proxies will never be non-configurable.